### DISCUSSION TITLE
Way to adjust transaction size when doing batch inserts/updates/deletes?

### DISCUSSION THREAD

**Discussion by ssoper-usgs** [FraunhoferIOSB/FROST-Server#1751] [Q&A]
Is there a setting to adjust transaction size when doing batch operations against the frost database? We're experimenting with a bulk data loading process and noticing that we encounter this particular issue at the moment: https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/apg-waits.xactsync.html.  One thing we could potentially play with is the size of the transaction before a commit is made.  Any ideas or tips would be much appreciated, thanks!

**Comment by hylkevds** [comment]
All requests in a batch with the same `atomicityGroup` are executed in a single transaction, so they either all succeed, or none. You could make the transactions smaller by making multiple groups, each with fewer requests. The disadvantage is that it's not possible to back-refer to items from other groups, since that group may have failed.
